centreon/www/front_src/src/Resources/Graph/Performance/TimePeriods/PopoverCustomTimePeriodPicker.tsx

```diff
@@ -0,0 +1,241 @@
+import { useTranslation } from 'react-i18next';
+import { makeStyles } from 'tss-react/mui';
+
+import { FormHelperText, Popover, Typography } from '@mui/material';
+import { LocalizationProvider } from '@mui/x-date-pickers';
+
+import { userAtom } from '@centreon/ui-context';
+
+import { CustomTimePeriodProperty } from '../../../Details/tabs/Graph/models';
+import {
+  labelEndDate,
+  labelEndDateGreaterThanStartDate,
+  labelFrom,
+  labelStartDate,
+  labelTo
+} from '../../../translatedLabels';
+import useDateTimePickerAdapter from '../../../useDateTimePickerAdapter';
+
+import DateTimePickerInput from './DateTimePickerInput';
+import {
+  CustomStyle,
+  OriginHorizontalEnum,
+  OriginVerticalEnum,
+  PickersData,
+  PopoverData,
+  anchorReferenceEnum
+} from './models';
+
+const useStyles = makeStyles()((theme) => ({
+  paper: {
+    '& .MuiPopover-paper': {
+      minWidth: 250,
+      padding: theme.spacing(1)
+    }
+  }
+}));
+
+const defaultPickersData = {
+  disabledPickerEndInput: false,
+  disabledPickerStartInput: false,
+  getIsErrorDatePicker: (): void => undefined,
+  maxDatePickerEndInput: undefined,
+  maxDatePickerStartInput: undefined,
+  minDatePickerEndInput: undefined,
+  minDatePickerStartInput: undefined,
+  onCloseEndPicker: (): void => undefined,
+  onCloseStartPicker: (): void => undefined
+};
+
+const defaultPopoverData = {
+  anchorEl: undefined,
+  anchorOrigin: {
+    horizontal: OriginHorizontalEnum.center,
+    vertical: OriginVerticalEnum.top
+  },
+  anchorPosition: undefined,
+  anchorReference: anchorReferenceEnum.none,
+  onClose: (): void => undefined,
+  transformOrigin: {
+    horizontal: OriginHorizontalEnum.center,
+    vertical: OriginVerticalEnum.top
+  }
+};
+
+const defaultCustomStyle = {
+  classNameError: undefined,
+  classNamePaper: undefined,
+  classNamePicker: undefined
+};
+
+export interface Props {
+  customStyle?: CustomStyle;
+  pickersData: PickersData;
+  popoverData: PopoverData;
+  renderBody?: JSX.Element;
+  renderFooter?: JSX.Element;
+  renderTitle?: JSX.Element;
+}
+
+const PopoverCustomTimePeriodPickers = ({
+  popoverData,
+  customStyle = defaultCustomStyle,
+  renderTitle,
+  renderBody,
+  renderFooter,
+  pickersData
+}: Props): JSX.Element => {
+  const { classes, cx } = useStyles();
+  const { t } = useTranslation();
+  const { classNamePaper, classNameError, classNamePicker } = customStyle;
+  const {
+    open,
+    onClose,
+    anchorOrigin,
+    transformOrigin,
+    anchorPosition,
+    anchorReference,
+    anchorEl
+  } = {
+    ...defaultPopoverData,
+    ...popoverData
+  };
+
+  const {
+    acceptDate,
+    customTimePeriod,
+    disabledPickerEndInput,
+    disabledPickerStartInput,
+    getIsErrorDatePicker,
+    maxDatePickerEndInput,
+    maxDatePickerStartInput,
+    minDatePickerEndInput,
+    minDatePickerStartInput,
+    onCloseEndPicker,
+    onCloseStartPicker
+  } = { ...defaultPickersData, ...pickersData };
```

Comment:
Some props in your DateTimePickerInput are optional. Don't override with default and irrelevant default props.

Reply:
I put default values ​​ to be able to use the destruction , and avoid the check , !isNil(pickersData ) && !isNil(pickersData?.acceptData) ? {acceptData}=pickersData ...

Reply:
Ok then why did you set those properties as optional in your components if you don't handle that they are "optional"? Either, you handle that or you set properties as mandatory

Reply:
bcs they are optional .. i see